libs/platform/router/src/services/router.service.ts

```diff
@@ -13,6 +13,7 @@ export interface RouterService {
   acceptParams(params: RouteParams): void;
   getUrl(route?: string, extras?: NavigationExtras): Observable<string>;
   getPathId(id: string): string | undefined;
+  shouldGuardRoute(): Observable<boolean>;
```

Comment:
I'm wondering why we need this public on the RouterService. 
Ideally router will handle navigation or not inside, depending on the leave calback result, without exposing anything.